sympy/polys/rootoftools.py

```diff
@@ -774,13 +774,58 @@ def _get_roots(cls, method, poly, radicals):
             if x.name not in free_names:
                 poly = poly.xreplace({d: x})
                 break
+
+        # compute rational lift if real algebraic
+        has_alg_coeff = any(c.is_algebraic and not c.is_rational
+                            for c in poly.all_coeffs())
+        has_alg_coeff = has_alg_coeff and all(c.is_real
+                            for c in poly.all_coeffs())
+
+        if has_alg_coeff:
+            poly_original = poly
+            poly = Poly(poly.expr, poly.gen, extension=True)
+            poly = poly.lift()
+
         coeff, poly = cls._preprocess_roots(poly)
         roots = []
 
         for root in getattr(cls, method)(poly):
             roots.append(coeff*cls._postprocess_root(root, radicals))
+
+        # filter out extraneous roots if introduced from lifting
+        if has_alg_coeff:
```

Comment:
Rather than checking this in multiple places throughout the function it would be better to have two methods `_get_roots_qq` and `_get_roots_alg`. Then dispatch on `poly.domain`:
```
if poly.domain.is_QQ:
    return cls._get_roots_qq(...)
elif poly.domain.is_AlgebraicField:
    return cls._get_roots_alg(...)
```